core/src/main/scala/akka/kafka/internal/ConsumerStage.scala

```diff
@@ -211,8 +211,13 @@ private[kafka] trait MessageBuilder[K, V, Msg] {
   def createMessage(rec: ConsumerRecord[K, V]): Msg
 }
 
-private[kafka] trait PromiseControl extends Control {
-  this: GraphStageLogic =>
+private[kafka] sealed trait ControlOperation
+
+private[kafka] case object ControlStop extends ControlOperation
+
+private[kafka] case object ControlShutdown extends ControlOperation
+
+private[kafka] trait PromiseControl extends GraphStageLogic with Control with CallbackWrapper[ControlOperation] {
```

Comment:
This is related because Akka 2.5.0 complains that `java.lang.IllegalStateException: not yet initialized: only setHandler is allowed in GraphStageLogic constructor` in some tests. Indeed, we expose callback to the outside, so if someone stops the stage before it gets fully initialized, it may throw such error. I guess 2.4.x wasn't checking this to such extent?